crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2111,6 +2151,12 @@ mod tests {
     #[test_case(Ty::Tuple(vec![Ty::IntLiteral(42), Ty::StringLiteral("foo")]), Ty::Tuple(vec![Ty::BuiltinInstance("int"), Ty::BuiltinInstance("str")]))]
     #[test_case(Ty::Tuple(vec![Ty::BuiltinInstance("int"), Ty::StringLiteral("foo")]), Ty::Tuple(vec![Ty::BuiltinInstance("int"), Ty::BuiltinInstance("str")]))]
     #[test_case(Ty::Tuple(vec![Ty::IntLiteral(42), Ty::BuiltinInstance("str")]), Ty::Tuple(vec![Ty::BuiltinInstance("int"), Ty::BuiltinInstance("str")]))]
+    #[test_case(Ty::Intersection{pos: vec![Ty::BuiltinInstance("int"), Ty::IntLiteral(2)], neg: vec![]}, Ty::IntLiteral(2))]
```

Comment:
I'm not sure which case this was intended to test, but I don't think it is testing intersection subtype-of logic; it is only testing that `int & Literal[2]` immediately simplifies to `Literal[2]`, because `Literal[2] <: int`. So then of course `Literal[2] <: Literal[2]`.

Reply:
Thanks! 
I wanted to test that not all positive intersection members must be subtypes of the target, only one is required.
Fixed in this commit - https://github.com/astral-sh/ruff/pull/14037/commits/1fd24024801cec1899640e7b611864ea6061ad2f